### DISCUSSION TITLE
How to do QoS for virtiofs

### DISCUSSION THREAD

**Discussion by HowHsu** [cloud-hypervisor/cloud-hypervisor#4554] [General]
I'm going to use virtiofs in cloudhypervisor.It seems the disk device support bandwidth control (through RateLimiterConfig ?) while
virtiofs cannot. How do we do something like bandwidth control for virtiofs device?

**Comment by sboeuf** [comment]
This isn't something that can be controlled at the VMM level for virtio-fs given `virtio-fs` is a `vhost-user` device. That means the rate limiting would have to happen from the backend `virtiofsd`. There was some discussions related to virtiofsd performance here: https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4387#issuecomment-1202374639
It mentions how you can increase the amount of threads to get better performance but I understand this is different from rate limiting.
I suggest you could start a discussion on the virtiofsd repository here: https://gitlab.com/virtio-fs/virtiofsd
